### ISSUE TITLE
Annotated optional structs loose their inner schemas

### ISSUE THREAD

**Issue by golergka** [jacob-alford/schemata-ts#265]
Reproduction:

```Typescript
const exampleSchema = S.Struct({
	annotatedOptionalStruct: S.Annotate({
		title: 'Annotated optional struct',
		description: 'Description of the annotated optional struct'
	})(S.Optional(S.Struct({ field: S.String })))
})

const jsonSchema = JS.stripIdentity(getJsonSchema(exampleSchema))
console.log(JSON.stringify(jsonSchema, null, 2))
```

This outputs:

```Typescript
{
  "properties": {
    "annotatedOptionalStruct": {
      "title": "Annotated optional struct",
      "description": "Description of the annotated optional struct"
    }
  },
  "required": [
    "annotatedOptionalStruct"
  ],
  "type": "object"
}
```

**Comment by golergka** [comment]
I've tried to change the order:

```Typescript
const exampleSchema = S.Struct({
	annotatedOptionalStruct: pipe(
		S.Struct({ field: S.String }),
		S.Annotate({
			title: 'Annotated optional struct',
			description: 'Description of the annotated optional struct'
		}),
		S.Optional
	)
})
```

But this makes things even worse:

```Typescript
{
  "properties": {
    "annotatedOptionalStruct": {}
  },
  "required": [
    "annotatedOptionalStruct"
  ],
  "type": "object"
}
```

**Comment by jacob-alford** [comment]
Ah, I see what the problem is.  It is fixed in version 2, but I will push a hotfix up today and get it released

**Comment by jacob-alford** [comment]
You can see the fix in this commit, with a test for this issue at the bottom:

https://github.com/jacob-alford/schemata-ts/pull/266/commits/a1bd1493bf218a92364373476f07470216c94f46

Once CI passes I will get this merged and version 1.4.3 released
